File: RecyclerView_3_MusicPlayer/app/src/main/java/com/example/recyclerview_3_musicplayer/MusicAdapter.java
package com.example.recyclerview_3_musicplayer;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class MusicAdapter extends RecyclerView.Adapter<MusicViewHolder> {

    private ClickListener clickListener;
    private ArrayList<Music> musicArrayList;

    public MusicAdapter(ArrayList<Music> music, ClickListener clickListener) {
        this.musicArrayList = music;
        this.clickListener = clickListener;
    }

    @NonNull
    @Override
    public MusicViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_layout, parent, false);
        return new MusicViewHolder(view, clickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull MusicViewHolder holder, int position) {
        Music music = musicArrayList.get(position);
        holder.setData(music);
    }



    @Override
    public int getItemCount() {
        return musicArrayList.size();
    }

    public void updateData(ArrayList<Music> musicArrayList) {
        this.musicArrayList = musicArrayList;
        notifyDataSetChanged();
    }
}

File: itunesApp/app/src/main/java/com/example/itunesapp/ui/MusicAdapter.kt
package com.example.itunesapp.ui

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.itunesapp.R
import com.example.itunesapp.data.model.MusicTable
import com.example.itunesapp.databinding.MusicItemLayoutBinding

class MusicAdapter(val list:ArrayList<MusicTable>) :RecyclerView.Adapter<MusicViewHolder>(){
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MusicViewHolder {
        return MusicViewHolder(DataBindingUtil.inflate(LayoutInflater.from(parent.context), R.layout.music_item_layout,parent,false))
    }

    override fun onBindViewHolder(holder: MusicViewHolder, position: Int) {
        val result = list[position]
        holder.setData(result)
    }

    override fun getItemCount(): Int {
        return list.size
    }
}

class MusicViewHolder(val musicItemLayoutBinding: MusicItemLayoutBinding):RecyclerView.ViewHolder(musicItemLayoutBinding.root){

    fun setData(result: MusicTable){
        musicItemLayoutBinding.result = result
        Glide.with(musicItemLayoutBinding.imgItem).load(result.artworkUrl30).into(musicItemLayoutBinding.imgItem)
    }
}